### ISSUE TITLE
Lutron Occupancy Sensor "illuminance"

### ISSUE THREAD

**Issue by YourMom11** [gurumitts/pylutron-caseta#83]
Is it technically possible to leverage the illuminance capability of the Lutron Occupancy family of Caseta/RA2 Select sensors? I would consider financially supporting the development if it's possible.

**Comment by mdonoughe** [comment]
It might be possible, but that information isn't contained in any of the test data and I don't have one to play with. Maybe there's a way to get it from the sensor? https://github.com/gurumitts/pylutron-caseta/blob/dev/tests/responses/occupancygroups.json#L50

@swails do you know?

**Comment by iCSpotRun** [comment]
Happy to send you a Sensor (to keep) if that helps. Let me know.

**Comment by iCSpotRun** [comment]
@mdonoughe @swails hope you had a chance to put more thought into this. Please let me know if sending you a sensor would be worthwhile. Cheers.

**Comment by mdonoughe** [comment]
Are you sure this sensor has this capability? I got a PD-OSENS-XX and it doesn't seem to report any status in the app or the API. There's an area setting for daylightgainsetting, but it seems to be for different hardware. Lutron does sell a light sensor but it's not a Caseta device. Maybe Lutron uses some of the same software for their other systems.

**Comment by iCSpotRun** [comment]
The **Radio Powr Savr wireless occupancy/vacancy sensors** for RA2 Select and RadioRA2 both support "Low Light Detection". Caseta versions do not. For RA2 sensors, this can only be enabled or disabled at the device.

**Description**
Low light: The lights will only turn on automatically upon entry if there is not already sufficient ambient light in the room.

The part number I have is: LRF2-OCR2B-P-WH
https://www.lutron.com/en-us/products/pages/wholehomesystems/ra2select/components.aspx#?cat=PC71

This makes me think that it can measure illuminance but I suspect it's not conveniently exposed via the API. I wonder if it can be accessed via Telnet. With RA2, Telnet can be enabled via the iOS app.

**Comment by swails** [comment]
Telnet is implemented via the `pylutron` library -- this one is LEAP-specific.  As @mdonoughe pointed out, the light-detection sensors are not available via Caseta.  It wouldn't surprise me if RA2 Select and RA3 both have hardware support.

That said, I've never interacted with those devices before and so don't know how this feature is implemented.  Given what I know about how these systems work, I *suspect* that this feature is implemented by adding logic to the sensor firmware that prevents it from sending an occupancy event except in low-light conditions (although I could very well be mistaken here).

An easy way to check it out is to fire up a LEAP and telnet listener and print out the messages that are broadcast by the bridge.

**Comment by mdonoughe** [comment]
LEAP isn't so simple. There are some implicitly subscribed events you get just for connecting, but for other events you need to subscribe to them first. If you have one connected you can try sending ReadRequests to the the device, the device "/status" and any URLs returned with those to see if there is a light level returned.

**Comment by swails** [comment]
Yes but we already subscribe to `/occupancygroup/status` which may yield messages about illumination settings if they actually come in separate LEAP messages.  We could also spam `SubscribeRequest` messages to various `/<something>/status` endpoints.

That said, if the low-light feature operates by *only* turning lights on in low light, I suspect it only broadcasts an occupied message if both the low-light condition *and* activity detected conditions are simultaneously true.

One way to test this would be to see if the occupancy group reports an occupied state when the sensor is placed in a well-lit area and you walk into it.  If the occupancy group *doesn't* report being occupied, I think that increases the likelihood you can't extract low-light measurements from the sensor in any way.
